api/client/authn.go

```diff
@@ -0,0 +1,108 @@
+package client
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/pkg/term"
+)
+
+// ValidateAuthnOpt checks if a passed-in option value is a recognized
+// authentication option.
+func ValidateAuthnOpt(option string) (string, error) {
+	if strings.HasPrefix(option, "basic.username=") ||
+		strings.HasPrefix(option, "basic.password=") ||
```

Comment:
I had this same concern in the design: I don't think we should allow passwords to be set as cmd-line arguments (I understand that `docker login` has the same problem, but we should definitely not propagate bad design decisions).